### PR TITLE
SDK-1076: Create task to expose `spryk-run`

### PR DESCRIPTION
**Pull request by Kyrylo-Meretskyi**
[SDK-1076](https://spryker.atlassian.net/browse/SDK-1076)

### CODE REVIEW

File: src/Infrastructure/Reader/TaskYamlReader.php [hunk 1]
```diff
@@ -75,6 +71,22 @@ public function readFiles(): ManifestCollectionDto
...
     }
 
+    /**
+     * @throws \SprykerSdk\Sdk\Core\Application\Exception\MissingSettingException
+     *
+     * @return \SprykerSdk\SdkContracts\Entity\SettingInterface
+     */
+    protected function getExtensionDirsSetting(): SettingInterface
+    {
+        $taskDirSetting = $this->settingRepository->findOneByPath(Setting::PATH_EXTENSION_DIRS);
```
Comment (vol4onok) [inline src/Infrastructure/Reader/TaskYamlReader.php:81]: you can use get method. it always present and remove exception

File: config/services_test.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+services:
+    task.reader.yaml:
+        class: SprykerSdk\Sdk\Infrastructure\Reader\TestTaskYamlReader
```
Comment (vol4onok) [inline config/services_test.yaml:3]: move it to test folder. maybe helper if you need to replace some service

File: tests/Sdk/Task/SprykRunTask.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Copied from Extension/Resources/config/task/SprykRunTask.yaml and changed type to `TYPE_LOCAL_CLI`,
+# because CI doesn't support `TYPE_LOCAL_CLI_INTERACTIVE` type.
+---
+id: "spryk:run:local"
```
Comment (vol4onok) [inline tests/Sdk/Task/SprykRunTask.yaml:4]: What is the idea of this test? as I see you don't test from an application. you test task from test env

File: src/Infrastructure/Reader/TestTaskYamlReader.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ */
+
+namespace SprykerSdk\Sdk\Infrastructure\Reader;
+
+use SprykerSdk\Sdk\Core\Application\Dependency\Repository\SettingRepositoryInterface;
+use SprykerSdk\SdkContracts\Entity\SettingInterface;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Yaml\Yaml;
+
+class TestTaskYamlReader extends TaskYamlReader
```
Comment (sergeyspryker) [inline src/Infrastructure/Reader/TestTaskYamlReader.php:15]: We don't a test scope here. Just move logic of paths fetching in separate class `SettingsPathProvider` and `StaticPathProvider` that just passes paths from argument. And separate test scope by a service as you've already done

File: src/Infrastructure/Service/TaskOptionBuilder.php [hunk 5]
```diff
@@ -104,10 +105,18 @@ protected function addPlaceholderOptions(TaskInterface $task, array $options): a
                 continue;
             }
 
+            $mode = $placeholder->isOptional() ? InputOption::VALUE_OPTIONAL : InputOption::VALUE_REQUIRED;
```
Comment (sergeyspryker) [inline src/Infrastructure/Service/TaskOptionBuilder.php:108]: Move mode building in separate method pls.

**Review by sergeyspryker — CHANGES_REQUESTED** [review]

File: src/Infrastructure/Reader/TaskYamlReader.php (hunk 1 above)
Comment (Kyrylo-Meretskyi) [reply src/Infrastructure/Reader/TaskYamlReader.php]: Fixed

File: config/services_test.yaml (hunk 2 above)
Comment (Kyrylo-Meretskyi) [reply config/services_test.yaml]: Fixed

File: tests/Sdk/Task/SprykRunTask.yaml (hunk 3 above)
Comment (Kyrylo-Meretskyi) [reply tests/Sdk/Task/SprykRunTask.yaml]: Fixed

File: src/Infrastructure/Reader/TestTaskYamlReader.php (hunk 4 above)
Comment (Kyrylo-Meretskyi) [reply src/Infrastructure/Reader/TestTaskYamlReader.php]: Fixed

File: src/Infrastructure/Service/TaskOptionBuilder.php (hunk 5 above)
Comment (Kyrylo-Meretskyi) [reply src/Infrastructure/Service/TaskOptionBuilder.php]: Fixed

File: tests/_project/project/src/Pyz/Glue/AuthRestApi/Object.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (sergeyspryker) [inline tests/_project/project/src/Pyz/Glue/AuthRestApi/Object.json:1]: After the test be finished this file become tracked by git as updated file. Need to put in gitignore and created it in the beginning of the test and remove after

**Review by sergeyspryker — CHANGES_REQUESTED** [review]

**Review by sergeyspryker — APPROVED** [review]

**Review by vol4onok — APPROVED** [review]

**Review by pavelmaksimov25 — APPROVED** [review]

**Pull request merged**